src/main/java/Sakura/Parser.java

```diff
@@ -0,0 +1,51 @@
+package Sakura;
+
+import java.io.IOException;
+
+/**
+ * Represents a parser for the user input
+ */
+public class Parser {
+    /**
+     * Takes in user input and executes further actions depending on the command word.
+     * Command words are bye, list, mark, unmark, todo, deadline, event, delete
+     *
+     * @param input string command given by user
+     * @param taskList Tasklist object created with Sakura
+     */
+    public static void parseCommand(String input, TaskList taskList) {
+        Sakura.ui.printDiv();
+        if (input.equals("bye")) {
+            Sakura.inProgress = false;
+            exit();
+        } else if (input.equals("list")) {
+            Ui.showAllTask(taskList.tasks);
+        } else if (input.toLowerCase().startsWith("mark")) {
+            taskList.markTask(input);
+        } else if (input.toLowerCase().startsWith("unmark")) {
+            taskList.unmarkTask(input);
+        } else if (input.toLowerCase().startsWith("todo") || input.toLowerCase().startsWith("deadline") || input.toLowerCase().startsWith("event")) {
+            taskList.addTask(input);
+        } else if (input.toLowerCase().startsWith("delete")) {
+            taskList.deleteTask(input);
+        } else if (input.toLowerCase().startsWith("find")) {
+            taskList.searchTask(input);
+        } else {
+            SakuraException.genericTask();
+        }

```

Comment:
Consider putting the toLowerCase() at the start by retrieving the first word in the input and using a switch statement. This makes it:
1. Easier to read.
2. Accept Bye and List commands which are not lowercase.